tests/preprocessing/test_triplegs.py

```diff
@@ -356,6 +356,31 @@ def test_only_staypoints_in_trip(self):
         assert (tpls_["trip_id"] == 0).all()
         assert len(trips) == 1
 
+    def test_loop_linestring_case(self):
+        # load pregenerated trips
+        trips_loaded = ti.read_trips_csv(os.path.join("tests", "data", "geolife_long", "trips.csv"), index_col="id")
+
+        # create trips from geolife (based on positionfixes)
+        pfs, _ = ti.io.dataset_reader.read_geolife(os.path.join("tests", "data", "geolife_long"))
+        pfs, stps = pfs.as_positionfixes.generate_staypoints(
+            method="sliding", dist_threshold=25, time_threshold=5, gap_threshold=1e6
+        )
+        stps = stps.as_staypoints.create_activity_flag(time_threshold=15)
```

Comment:
I think we can delete the activity flag

Reply:
This gives an error ```AssertionError: staypoints need the column 'activities' to be able to generate trips```
So I think we need to keep that